librz/main/rz-asm.c

```diff
@@ -168,6 +168,110 @@ static int show_analinfo(RzAsmState *as, const char *arg, ut64 offset) {
 	return ret;
 }
 
+static const char *has_esil(RzAsmState *as, const char *name) {
+	RzListIter *iter;
+	RzAnalysisPlugin *h;
+	rz_list_foreach (as->analysis->plugins, iter, h) {
+		if (h->name && !strcmp(name, h->name)) {
+			return h->esil ? "Ae" : "A_";
+		}
+	}
+	return "__";
+}
+
+static void rz_asm_list(RzAsmState *as, const char *arch) {
+	int i;
+	char bits[32];
+	const char *feat2, *feat;
+	RzAsmPlugin *h;
+	RzListIter *iter;
+	PJ *pj = pj_new();
+	if (!pj) {
+		return;
+	}
+	if (as->json) {
+		pj_o(pj);
+	}
+	rz_list_foreach (as->a->plugins, iter, h) {
+		if (arch) {
+			if (h->cpus && !strcmp(arch, h->name)) {
+				char *c = strdup(h->cpus);
+				int n = rz_str_split(c, ',');
+				for (i = 0; i < n; i++) {
+					printf("%s\n", rz_str_word_get0(c, i));
+				}
+				free(c);
+				break;
+			}
+		} else {
+			bits[0] = 0;
+			if (h->bits == 27) {
+				strcat(bits, "27");
+			} else if (h->bits == 0) {
+				strcat(bits, "any");
+			} else {
+				if (h->bits & 4) {
+					strcat(bits, "4 ");
+				}
+				if (h->bits & 8) {
+					strcat(bits, "8 ");
+				}
+				if (h->bits & 16) {
+					strcat(bits, "16 ");
+				}
+				if (h->bits & 32) {
+					strcat(bits, "32 ");
+				}
+				if (h->bits & 64) {
+					strcat(bits, "64 ");
+				}
+			}
+			feat = "__";
+			if (h->assemble && h->disassemble) {
+				feat = "ad";
+			}
+			if (h->assemble && !h->disassemble) {
+				feat = "a_";
+			}
+			if (!h->assemble && h->disassemble) {
+				feat = "_d";
+			}
+			feat2 = has_esil(as, h->name);
+			if (as->quiet) {
+				printf("%s\n", h->name);
+			} else if (as->json) {
+				pj_k(pj, h->name);
+				pj_o(pj);
+				pj_k(pj, "bits");
+				pj_a(pj);
+				pj_i(pj, 32);
+				pj_i(pj, 64);
```

Comment:
here, why you hardcode 32 and 64?

Reply:
yea this is in master, I just reverted to it because my changes were not working as intended.